### PR TITLE
feat: Stride <> Eclipse TIA and stTIA

### PR DESCRIPTION
**Pull request by daniel-savu**

### CODE REVIEW

File: rust/sealevel/environments/mainnet3/eclipsemainnet/gas-oracle-configs-eclipsemainnet.json [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+      "gasPrice": "15",
+      "tokenDecimals": 9
+    }
+  },
+  {
+    "domain": 745,
+    "gasOracle": {
+      "type": "remoteGasData",
+      "tokenExchangeRate": "46951091051",
+      "gasPrice": "0.025",
```
Comment (tkporter) [inline rust/sealevel/environments/mainnet3/eclipsemainnet/gas-oracle-configs-eclipsemainnet.json:25]: I think gasPrice must be an integer so we need to do some precision adjustments

Using a test I came up with these numbers, wdyt?

```
    #[test]
    fn test_quote_gas_payment() {
        let mut igp = Igp::default();
        let stride_domain = 745;

        igp.gas_oracles.insert(
            stride_domain,
            GasOracle::RemoteGasData(RemoteGasData {
                // ((0.7331 / 2526.50) * 1e19 (exchange rate scale) * 1.5 (apply premium)) / 10 (scaling down to let the gas price have more precision)
                token_exchange_rate: 435246388284187,
                // Scaling up the gas price to reach a desired cost
                gas_price: 7,
                token_decimals: 6,
            }),
        );

        let quote = igp.quote_gas_payment(stride_domain, 300000).unwrap();
        println!("quote: {}", quote);
  }
```

which gives a quote of `91000` eth lamports, which is `(91000 / 1e9) * 2526 = 23 cents`

File: typescript/infra/config/environments/mainnet3/warp/warpIds.ts [hunk 2]
```diff
@@ -17,4 +17,6 @@ export enum WarpRouteIds {
   EthereumZircuitPZETH = 'PZETH/ethereum-zircuit',
   InevmInjectiveINJ = 'INJ/inevm-injective',
   MantapacificNeutronTIA = 'TIA/mantapacific-neutron',
+  StrideEclipseTIA = 'TIA/stride-eclipse',
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/warp/warpIds.ts:20]: nit: I think the chains need to be alphabetically sorted, at least thats how the warp ID generation logic works, we should try to be consistent

**Review by tkporter — APPROVED** [review]

File: typescript/infra/config/environments/mainnet3/warp/warpIds.ts (hunk 2 above)
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/warp/warpIds.ts:20]: yup it's `<TICKER>/alphabet-chain-order`

File: typescript/infra/config/environments/mainnet3/warp/warpIds.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ export enum WarpRouteIds {
   ArbitrumNeutronTIA = 'TIA/arbitrum-neutron',
   EclipseSolanaSOL = 'SOL/eclipsemainnet-solanamainnet',
   EclipseSolanaWIF = 'WIF/eclipsemainnet-solanamainnet',
+  EclipseStrideSTTIA = 'STTIA/eclipse-stride',
```
Comment (Mo-Hussain) [inline typescript/infra/config/environments/mainnet3/warp/warpIds.ts:9]: I think setting this value to stTIA/eclipse-stride should allow the checker to find the addresses in registry and solve your problem

File: typescript/infra/config/environments/mainnet3/warp/TIA-eclipsestride-deployments.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Configs and artifacts for the deployment of Hyperlane Warp Routes
+description: Hyperlane Warp Route artifacts
+timestamp: '2024-10-18T14:00:00.000Z'
+deployer: Abacus Works (Hyperlane)
+data:
+  config:
+    stride:
+      protocolType: cosmos
+      type: collateral
+      hypAddress: stride134axwdlam929m3mar3wv95nvkyep7mr87ravkqcpf8dfe3v0pjlqwrw6ee
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/warp/TIA-eclipsestride-deployments.yaml:10]: I think this is the stTIA address, should be `stride1pvtesu3ve7qn7ctll2x495mrqf2ysp6fws68grvcu6f7n2ajghgsh2jdj6`

**Pull request merged**